src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1061,12 +1065,31 @@ private Task<Stream> InternalGetRequestStream()
                 throw new ProtocolViolationException(SR.net_nouploadonget);
             }
 
+            if (AllowWriteStreamBuffering is false && SendChunked is false && ContentLength < 0)
+            {
+                throw new ProtocolViolationException(SR.net_nobuffering_nochunked);
+            }
+
             if (RequestSubmitted)
             {
                 throw new InvalidOperationException(SR.net_reqsubmitted);
             }
 
-            _requestStream = new RequestStream();
+            // Create stream buffer for transferring data from RequestStream to the StreamContent.
+            StreamBuffer streamBuffer = new StreamBuffer(
+                maxBufferSize: AllowWriteStreamBuffering is false ? int.MaxValue : StreamBuffer.DefaultMaxBufferSize);
```

Comment:
```suggestion
            StreamBuffer streamBuffer = new StreamBuffer(maxBufferSize: AllowWriteStreamBuffering ? int.MaxValue : StreamBuffer.DefaultMaxBufferSize);
```
The condition seems to be inversed.